web/src/components/auth/confirm-user/ConfirmUserForm.tsx

```diff
@@ -0,0 +1,142 @@
+
+const DEFAULT_VALUES: ConfirmUserRequest = {
+    code: '',
+    username: '',
+};
+
+const useAlert = () => {
+    return (message: string, type: string) => {
+        console.log(message, type);
+    };
+};
+
+const ConfirmUserForm = () => {
+    const router = useRouter();
+    const showAlert = useAlert();
+    const [username, setUsername] = useState<string>('');
+    const { confirmUser } = useContext(AuthContext);
+    const { formState, register, handleSubmit } = useForm<ConfirmUserRequest>(
+        USE_FORM_CONFIG<ConfirmUserRequest>(DEFAULT_VALUES, confirmUserSchema),
+    );
+    const { isValid, isSubmitting, isDirty, errors } = formState;
+
+    useEffect(() => {
+        setUsername(getItem('username') ?? '');
+    }, []);
+
+    if (!username) {
+        return <>We forgot your username... awkward.</>;
+    }
+
+    const onSubmit = async (request: ConfirmUserRequest) => {
+        try {
+            await confirmUser(request);
+            router.push('/auth/login');
+            return;
+        } catch (error: unknown) {
+            if (isAWSError(error, 'CodeMismatchException')) {
+                showAlert('Incorrect code.', 'error');
+            } else if (isAWSError(error, 'ExpiredCodeException')) {
+                showAlert('The code you entered has expired.', 'error');
+            } else if (
+                isAWSError(error, 'InvalidLambdaResponseException') ||
+                isAWSError(error, 'UnexpectedLambdaException') ||
+                isAWSError(error, 'UserLambdaValidationException')
+            ) {
+                showAlert(
+                    'An error occurred while adding user to database. Please contact support.',
+                    'error',
+                );
+            } else if (isAWSError(error, 'InvalidParameterException')) {
+                showAlert((error as Error).message, 'error');
+            } else if (isAWSError(error, 'TooManyFailedAttemptsException')) {
+                showAlert((error as Error).message, 'error');
+            } else if (isAWSError(error, 'TooManyRequests')) {
+                showAlert(TOO_MANY_REQUESTS(), 'error');
+            } else if (error instanceof Error) {
+                showAlert(error.message, 'error');
+            } else {
+                showAlert(UNKNOWN(), 'error');
+            }
+        }
+    };
+
+    const onCancel = () => {
+        router.back();
+    };
+
+    return (
+        <Box className='flex flex-col'>
+            <form
+                onSubmit={handleSubmit((data) =>
+                    onSubmit({ ...data, username }),
+                )}
+                noValidate
+            >
+                <Typography variant='h3'>Confirm Email</Typography>
+                <Typography variant='body1'>
+                    We&apos;ve sent an email with a verification code. Please
+                    enter it here.
+                </Typography>
+                <FormGroup className='mt-4'>
+                    <TextField
```

Comment:
Eventually, we will want to create a better component for this, which a textbox for each number, but this works for now.

Reply:
Created an issue for this #54